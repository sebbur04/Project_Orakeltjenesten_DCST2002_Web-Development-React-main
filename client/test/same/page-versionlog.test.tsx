// Import neccessary modules for testing
import * as React from "react";
import { shallow } from "enzyme";
import VersionLog from "../../src/same/page-versionlog";

jest.mock("../../src/service/versionlog-service", () => ({
  getAllPageVersions: jest.fn(() =>
    Promise.resolve([
      {
        id: 1,
        version: 1,
        pageid: 123,
        userid: 1,
        date: "2024-11-16T10:00:00Z",
        changelog: "Initial creation",
      },
      {
        id: 2,
        version: 2,
        pageid: 123,
        userid: 2,
        date: "2024-11-16T12:00:00Z",
        changelog: "Updated content",
      },
    ])
  ),
}));

jest.mock("../../src/service/page-service", () => ({
  getPage: jest.fn(() =>
    Promise.resolve({ id: 123, name: "Test Page", num_views: 100 })
  ),
}));

jest.mock("../../src/service/user-service", () => ({
  getAllUsers: jest.fn(() =>
    Promise.resolve([
      { id: 1, username: "user1", avatar: "", bio: "", permid: 0 },
      { id: 2, username: "user2", avatar: "", bio: "", permid: 0 },
    ])
  ),
}));

describe("VersionLog Component Tests", () => {
  test("Renders version logs correctly", (done) => {
    const match = { params: { id: 123 } };
    const wrapper = shallow(<VersionLog match={match} />);

    setTimeout(() => {
      // Check that the Card title renders correctly
      expect(wrapper).toMatchSnapshot();

      done();
    });
  });
});
