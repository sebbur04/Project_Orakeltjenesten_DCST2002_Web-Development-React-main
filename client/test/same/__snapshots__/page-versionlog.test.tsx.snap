// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VersionLog Component Tests Renders version logs correctly 1`] = `
<div
  className="mainContent"
>
  <Card
    title="Versjonslogg for "Test Page""
  >
    <Row
      key="1"
    >
      <Qe>
        ---
      </Qe>
      <Column
        width={2}
      >
        <a
          className="blue-link"
          href="/#/pages/123/versionlog/1"
        >
          Versjon 1
        </a>
      </Column>
      <Column
        width={2}
      >
        Endret av: 
        <a
          className="blue-link"
          href="/#/users/1"
        >
          user1
        </a>
      </Column>
      <Column
        width={3}
      >
        Dato: 2024-11-16 11:00
      </Column>
      <Column
        width={5}
      >
        Logg: Initial creation
      </Column>
    </Row>
    <Row
      key="2"
    >
      <Qe>
        ---
      </Qe>
      <Column
        width={2}
      >
        <a
          className="blue-link"
          href="/#/pages/123/versionlog/2"
        >
          Versjon 2
        </a>
      </Column>
      <Column
        width={2}
      >
        Endret av: 
        <a
          className="blue-link"
          href="/#/users/2"
        >
          user2
        </a>
      </Column>
      <Column
        width={3}
      >
        Dato: 2024-11-16 13:00
      </Column>
      <Column
        width={5}
      >
        Logg: Updated content
      </Column>
    </Row>
  </Card>
</div>
`;
